src/plugins/ui_actions/public/types.ts

```diff
@@ -51,6 +52,10 @@ export interface VisualizeFieldContext {
   contextualFields?: string[];
 }
 
+export interface ExternalActionContext {
+  data: any;
```

Comment:
for clarity - can we define this more to be `data: { event: any, datum: any }`? Since this action encompasses both click and mouse events, I think it's ok to be this fine-grained. And it makes it more clear when defining an action, what the context is that is passed, since it will have to perform differently based on the `event`.